internal/provider/serverless_cluster_resource.go

```diff
@@ -0,0 +1,730 @@
+				MarkdownDescription: "The endpoints for connecting to the cluster.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"public_endpoint": schema.SingleNestedAttribute{
+						MarkdownDescription: "The public endpoint for connecting to the cluster.",
+						Optional:            true,
+						Computed:            true,
+						Attributes: map[string]schema.Attribute{
+							"host": schema.StringAttribute{
+								MarkdownDescription: "The host of the public endpoint.",
+								Computed:            true,
+							},
+							"port": schema.Int64Attribute{
+								MarkdownDescription: "The port of the public endpoint.",
+								Computed:            true,
+							},
+							"disabled": schema.BoolAttribute{
+								MarkdownDescription: "Whether the public endpoint is disabled.",
+								Optional:            true,
+							},
+						},
+					},
+					"private_endpoint": schema.SingleNestedAttribute{
+						MarkdownDescription: "The private endpoint for connecting to the cluster.",
+						Computed:            true,
+						PlanModifiers: []planmodifier.Object{
+							objectplanmodifier.UseStateForUnknown(),
+						},
+						Attributes: map[string]schema.Attribute{
+							"host": schema.StringAttribute{
+								MarkdownDescription: "The host of the private endpoint.",
+								Computed:            true,
+							},
+							"port": schema.Int64Attribute{
+								MarkdownDescription: "The port of the private endpoint.",
+								Computed:            true,
+							},
+							"aws_endpoint": schema.SingleNestedAttribute{
+								MarkdownDescription: "Message for AWS PrivateLink information.",
+								Computed:            true,
+								Attributes: map[string]schema.Attribute{
+									"service_name": schema.StringAttribute{
+										MarkdownDescription: "The AWS service name for private access.",
+										Computed:            true,
+									},
+									"availability_zone": schema.ListAttribute{
+										MarkdownDescription: "The availability zones that the service is available in.",
+										Computed:            true,
+										ElementType:         types.StringType,
+									},
+								},
+							},
+							"gcp_endpoint": schema.SingleNestedAttribute{
+								MarkdownDescription: "Message for GCP PrivateLink information.",
+								Computed:            true,
+								Attributes: map[string]schema.Attribute{
+									"service_attachment_name": schema.StringAttribute{
+										MarkdownDescription: "The target GCP service attachment name for private access.",
+										Computed:            true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"root_password": schema.StringAttribute{
+				MarkdownDescription: "The root password of the cluster.",
+				Optional:            true,
+				Sensitive:           true,
```

Comment:
consider using writeonly? I does not use it before, maybe we can have a try.
https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#writeonly

Reply:
it's not supported in framework, and I'm using cockroach as my reference https://github.com/cockroachdb/terraform-provider-cockroach/blob/6a0a2843d682e3bfbaf9e8eccc4819fdcbbbc55c/internal/provider/sql_user_resource.go#L70